### PR TITLE
Overlay/underlay tiebreaker, `FLOAT_LAYER`, & Image over/underlays fix

### PR DESCRIPTION
**Pull request by amylizzle**
Basically replaces the previous attempt with a tree traversal algorithm that should preserve proper underlay and overlay render order even when there are multiply nested under/overlays. Also adds another visual test to ensure that they're working right. 
Also implements `FLOAT_LAYER` and makes `/image` work right when used as overlays.
image

### CODE REVIEW

**Comment by out-of-phaze** [comment]
With this PR merged, instead of the ordering on human mobs on Nebula being broken some of the time, it's broken all of the time. :(

**Comment by out-of-phaze** [comment]
Tested again—after removing a bad Overlays/Underlays.Sort() call outside of renderer code, it's... exactly as broken with or without this PR. Darn.

**Comment by amylizzle** [comment]
Works good now, if there's still a problem I think it must be something other than the ordering.

**Comment by amylizzle** [comment]
also note: this is copy pasted code from atom overlays. It's gross I know, but it should be a one-for-one swap-out when someone does overlay lists properly.

File: DMCompiler/DMStandard/Types/Image.dm [hunk 1]
```diff
@@ -1,17 +1,17 @@
 ﻿/image
 	parent_type = /datum
 
-	var/alpha
+	var/alpha = 255
```
Comment (wixoaGit) [inline DMCompiler/DMStandard/Types/Image.dm:4]: The defaults for `/image` don't come from here, they're whatever the default value in `IconAppearance` is.

I suppose setting it here is desirable for `initial()` though. Maybe leave a comment somewhere clarifying where the actual defaults come from?

Comment (amylizzle) [reply DMCompiler/DMStandard/Types/Image.dm:4]: So I should set these values in `IconAppearance` and `/image`?

Comment (amylizzle) [reply DMCompiler/DMStandard/Types/Image.dm:5]: These values are all already set in IconAppearance, so I just added the comment.

**Comment by ike709** [comment]
CI fails.

**Comment by amylizzle** [comment]
Finally got around to testing this - I can't replicate missing player sprites
image

**Comment by out-of-phaze** [comment]
> Finally got around to testing this - I can't replicate missing player sprites image

Yeah, player sprites work fine on Nebula now, I think? The issue was some weirdness with something like a blank, no-args image inheriting icon from its overlay, so it was overlaying a duplicate torso on top of the correctly-generated player sprite.

File: OpenDreamClient/Rendering/DreamViewOverlay.cs [hunk 2]
```diff
@@ -481,28 +488,30 @@ internal sealed class DreamViewOverlay : Overlay {
...
-                current.KeepTogetherGroup.AddRange(ProcessIconComponents(underlay, current.Position, uid, isScreen, current, keepTogether, underlayTiebreaker));
+                current.KeepTogetherGroup.AddRange(ProcessIconComponents(underlay, current.Position, uid, isScreen, ref tieBreaker, current, keepTogether));
             }
         }
 
+        tieBreaker++;
+        current.TieBreaker = tieBreaker;
+
+
         //overlays - colour, alpha, and transform are inherited, but filters aren't
```
Comment (wixoaGit) [inline OpenDreamClient/Rendering/DreamViewOverlay.cs:506]:
```suggestion
        tieBreaker++;
        current.TieBreaker = tieBreaker;

        //overlays - colour, alpha, and transform are inherited, but filters aren't
```

**Review by wixoaGit — CHANGES_REQUESTED** [review]

**Review by wixoaGit — APPROVED** [review]

**Pull request merged**